engine/transaction/record/record_outline.go

```diff
@@ -0,0 +1,126 @@
+package record
+
+import (
+	"context"
+
+	"github.com/bitcoin-sv/go-sdk/spv"
+	trx "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/conv"
+	"github.com/bitcoin-sv/spv-wallet/engine/database"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/transaction/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/transaction/outlines"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+// RecordTransactionOutline will validate, broadcast and save a transaction outline
+func (s *Service) RecordTransactionOutline(ctx context.Context, outline *outlines.Transaction) error {
+	tx, err := trx.NewTransactionFromBEEFHex(outline.BEEF)
+	if err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	}
+
+	if ok, err := spv.VerifyScripts(tx); err != nil {
+		return txerrors.ErrTxValidation.Wrap(err)
+	} else if !ok {
+		return txerrors.ErrTxValidation
+	}
+
+	utxos, err := s.getTrackedUTXOsFromInputs(ctx, tx)
+	if err != nil {
+		return err
+	}
+
+	txID := tx.TxID().String()
+	for _, utxo := range utxos {
+		utxo.Spend(txID)
+	}
+
+	var newOutputs []*database.Output
+	var newDataRecords []*database.Data
+	if outline.Annotations != nil {
+		newOutputs, newDataRecords, err = s.processAnnotatedOutputs(tx, *outline.Annotations)
+		if err != nil {
+			return err
+		}
+	}
+
+	if err = s.broadcaster.Broadcast(ctx, tx); err != nil {
+		return txerrors.ErrTxBroadcast.Wrap(err)
+	}
+
+	txRecord := database.Transaction{
+		ID:       txID,
+		TxStatus: database.TxStatusBroadcasted,
+	}
+
+	upsertOutputs := append(newOutputs, utxos...) //newly created outputs and spent utxos
+	err = s.repo.SaveTX(ctx, &txRecord, upsertOutputs, newDataRecords)
+	if err != nil {
+		return txerrors.ErrSavingData.Wrap(err)
+	}
+
+	return nil
+}
+
+func (s *Service) getTrackedUTXOsFromInputs(ctx context.Context, tx *trx.Transaction) ([]*database.Output, error) {
+	outpoints := func(yield func(outpoint bsv.Outpoint) bool) {
```

Comment:
This doesn't need another "temporary" slice - it is a new "cool" feature in golang - I would give it a change - because it has some benefits (sometimes).
- Or I overuse it 🙈 
- But in this case I would leave it :)

Reply:
Personal opinion (maybe I need to convince myself to this syntax) - it looks a bit overwhelming 🙈